docs/wallet-integration-guide/src/exchange-integration/extensions.rst

```diff
@@ -6,16 +6,94 @@ Integration Extensions
 Multi-Hosted Parties
 ~~~~~~~~~~~~~~~~~~~~
 
-Sketch:
-
-* allocate the ``treasuryParty`` from the start as a `decentralized party <https://docs.digitalasset.com/operate/3.4/howtos/operate/parties/decentralized_parties.html>`__ on multiple validator nodes
-
-  * a common setup is to use two nodes with threshold two; thus both of them must confirm a tx for it to be committed
-
-* upload all .dars for your onboarded tokens on all of these validators
-* use any of the nodes hosting the party to read from as part of history ingestion and to prepare and execute transactions
-
-  * switching between the nodes can be done. take care to re-synchronize the ingestion offset as described in :ref:`restore-from-validator-node-backup`
+As part of setting up the ``treasuryParty``, you need to sign the
+``PartyToParticipant`` mapping which defines the set of confirming
+nodes for that party as well as the confirmation threshold. The
+confirming nodes for the party are essential to keep your party secure
+and compromise of them could lead to loss of funds. Refer to the trust
+model `trust model
+<https://docs.digitalasset.com/overview/3.3/explanations/canton/external-party.html#party-trust-model>`_
+for more details.
+
+To guard against compromise, you can setup your party with multiple
+confirming nodes and a threshold N > 1. As long as less than N nodes
+are compromised, your party is still secured. Common setups are:
+
+1. 2 confirming nodes with a threshold of 2. This provides security
+   against a single node being compromised. Note that it does not
+   provide availability in case of compromise though: If one node is
+   compromised or down, transactions will be rejected as you will not
+   be able to reach 2 valid confirmations. To be able to run
+   transactions again you either need to bring both nodes back online
+   or change the hosting nodes.
+2. 3 confirming nodes with a threshold of 2. This extends the previous
+   setup to also provide availability in case one of the nodes goes
+   down or gets compromised as you can still reach 2 valid
+   confirmations.
+
+Party Setup
+^^^^^^^^^^^
```

Comment:
This section is horrible. But we don't have wallet SDK support yet nor docs for the new Canton endpoints. And trying to work around that here feels like a bad investment so I'd rather keep it as is and make sure we get wallet SDK support sooner rather than later.

Reply:
Agreed. Let's add a ticket to change the docs once the base docs are available on the Canton side.

@daravep will you create docs for the new party hosting endpoints?